tests/AutomaticModeTest.php

```diff
@@ -71,17 +71,71 @@ public function context_is_switched_when_tenancy_is_reinitialized()
 
         $this->assertSame('foobar', app('tenancy_initialized_for_tenant'));
     }
+
+    /** @test */
+    public function running_the_global_tenancy_helper_with_tenant_already_initialized()
+    {
+        MyBootstrapper::$revertedCallCount = 0;
+        GlobalRun::$count = 0;
+
+        config(['tenancy.bootstrappers' => [
+            MyBootstrapper::class,
+        ]]);
+
+        $tenant = Tenant::create([
+            'id' => 'acme',
+        ]);
+
+        tenancy()->initialize($tenant);
+
+        $this->assertSame('acme', app('tenancy_initialized_for_tenant'));
+
+        tenancy()->runGlobal(function () {
+            GlobalRun::$count = 1;
```

Comment:
I think you could just assert that the connection was `tenant` before this call and is not `tenant` inside it.

Reply:
This is to make sure that the callback is indeed being called.

Reply:
Count is not really an intuitive way of testing things. I'd make assertions that:

1. The connection **was** `tenant` but is **not** anymore, *testing that we're in the central context*
2. Setting a value in `app()`, *testing that the callback is being called*

Reply:
Can you take a look at the changes to the tests and se if you like this better?